integrationTests/vm/wasm/queries/queries_test.go

```diff
@@ -0,0 +1,206 @@
+	network.Continue(t, 1)
+
+	// Block 1
+
+	now := queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{})
+	snapshotsOfGetNow[1] = now
+	network.Continue(t, 1)
+
+	// Block 2
+
+	now = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{})
+	snapshotsOfGetNow[2] = now
+	setState(network, scAddress, scOwner, 42, &scOwnerNonce)
+	network.Continue(t, 1)
+
+	// Block 3
+
+	state := getState(t, network.Node, scAddress, core.OptionalUint64{})
+	snapshotsOfGetState[3] = state
+	now = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{})
+	snapshotsOfGetNow[3] = now
+	setState(network, scAddress, scOwner, 43, &scOwnerNonce)
+	network.Continue(t, 1)
+
+	// Block 4
+
+	state = getState(t, network.Node, scAddress, core.OptionalUint64{})
+	snapshotsOfGetState[4] = state
+	now = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{})
+	snapshotsOfGetNow[4] = now
+	network.Continue(t, 1)
+
+	// Check snapshots
+	block1, _ := network.Node.GetShardHeader(1)
+	block2, _ := network.Node.GetShardHeader(2)
+	block3, _ := network.Node.GetShardHeader(3)
+	block4, _ := network.Node.GetShardHeader(4)
+
+	require.Equal(t, uint64(1), snapshotsOfGetNow[1].blockNonce)
+	require.Equal(t, uint64(2), snapshotsOfGetNow[2].blockNonce)
+	require.Equal(t, uint64(3), snapshotsOfGetNow[3].blockNonce)
+	require.Equal(t, uint64(4), snapshotsOfGetNow[4].blockNonce)
+
+	require.Equal(t, block1.GetRootHash(), snapshotsOfGetNow[1].stateRootHash)
+	require.Equal(t, block1.GetRootHash(), snapshotsOfGetNow[2].stateRootHash)
+	require.NotEqual(t, block2.GetRootHash(), snapshotsOfGetNow[3].stateRootHash)
+	require.NotEqual(t, block3.GetRootHash(), snapshotsOfGetNow[4].stateRootHash)
+
+	require.Equal(t, 42, snapshotsOfGetState[3])
+	require.Equal(t, 43, snapshotsOfGetState[4])
+
+	// Check history
+	historyOfGetState[1] = getState(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 1})
+	historyOfGetNow[1] = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 1})
+
+	historyOfGetState[2] = getState(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 2})
+	historyOfGetNow[2] = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 2})
+
+	historyOfGetState[3] = getState(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 3})
+	historyOfGetNow[3] = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 3})
+
+	historyOfGetState[4] = getState(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 4})
+	historyOfGetNow[4] = queryHistoryGetNow(t, network.Node, scAddress, core.OptionalUint64{HasValue: true, Value: 4})
+
+	require.Equal(t, snapshotsOfGetState[1], historyOfGetState[1])
+	require.Equal(t, snapshotsOfGetNow[1].blockNonce, historyOfGetNow[1].blockNonce)
+	// This does not seem right!
```

Comment:
Though, I think the issue is with the test node, not with the actual implementation.